test/parallel/test-crypto-dh.js

```diff
@@ -307,6 +308,29 @@ if (availableCurves.has('prime256v1') && availableCurves.has('secp256k1')) {
   });
 }
 
+// Use of invalid keys was not cleaning up ERR stack, and was causing
+// unexpected failure in subsequent signing operations.
+if (availableCurves.has('prime256v1') && availableHashes.has('sha256')) {
+  const curve = crypto.createECDH('prime256v1');
+  const invalidKey = Buffer.alloc(65);
+  invalidKey.fill('\0');
+  curve.generateKeys();
+  try {
+    curve.computeSecret(invalidKey);
+    assert(false, 'key generation should have failed');
+  } catch (err) {
+    // Check that signing operations are not impacted by the above error.
+    const ecPrivateKey =
+      '-----BEGIN EC PRIVATE KEY-----\n' +
+      'MHcCAQEEIF+jnWY1D5kbVYDNvxxo/Y+ku2uJPDwS0r/VuPZQrjjVoAoGCCqGSM49\n' +
+      'AwEHoUQDQgAEurOxfSxmqIRYzJVagdZfMMSjRNNhB8i3mXyIMq704m2m52FdfKZ2\n' +
+      'pQhByd5eyj3lgZ7m7jbchtdgyOF8Io/1ng==\n' +
+      '-----END EC PRIVATE KEY-----';
+    assert.ok(crypto.createSign('SHA256').sign(ecPrivateKey),
+              'signing should not throw');
```

Comment:
(Although I guess that doesn't check that you get a truthy return value. Could be added as an assertion to the function if it's important.)